controller/collaborators.go

```diff
@@ -114,9 +127,102 @@ func (c *CollaboratorsController) List(ctx *app.ListCollaboratorsContext) error
 	})
 }
 
+func (c *CollaboratorsController) listRoles(ctx *app.ListCollaboratorsContext, currentIdentity *account.Identity, roleName string, isServiceAccount bool) ([]rolerepo.IdentityRole, error) {
+	if isServiceAccount {
+		return c.app.IdentityRoleRepository().FindIdentityRolesByResourceAndRoleName(ctx, ctx.SpaceID.String(), roleName, false)
+	}
+	return c.app.RoleManagementService().ListByResourceAndRoleName(ctx, currentIdentity.ID, ctx.SpaceID.String(), roleName)
+}
+
+func (c *CollaboratorsController) listCollaborators(ctx *app.ListCollaboratorsContext, isServiceAccount bool) error {
+	var currentIdentity *account.Identity
+	var err error
+	if !isServiceAccount {
+		currentIdentity, err = login.LoadContextIdentityIfNotDeprovisioned(ctx, c.app)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+	}
+
+	cm := make(map[uuid.UUID]account.Identity) // Use map to check duplications
+	var collaborators []account.Identity
+	// Collect all contributors and admins of the space
+	contributors, err := c.listRoles(ctx, currentIdentity, authorization.SpaceContributorRole, isServiceAccount)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+	for _, c := range contributors {
+		cm[c.IdentityID] = c.Identity
+		collaborators = append(collaborators, c.Identity)
+	}
+	admins, err := c.listRoles(ctx, currentIdentity, authorization.SpaceAdminRole, isServiceAccount)
```

Comment:
When in the future we have people assigned to view role we won't use collaborators API anymore.

Reply:
OK in that case, you could use `roleManagementService().ListByResource(..)` and get all assignments instead of doing it once for admin, and another time for contributors.

Reply:
OK. Let me clarify what I meant :)
Yes, currently I want to return only admins and contributors as collaborators. It's technically possible for admins to add viewers though if using assign role API directly instead of collaborators API.
But when we start using assign API in our UI then it won't matter anymore because we will remove the collaborator API.